### PR TITLE
feat(wuc): extract order cart service

### PR DESCRIPTION
**Pull request by JeremyDec**
Extract of #1836 /packages/manager/apps/web/client/app/components/manager-order-service/manager-order-service.service.js

Name of service changed into wucOrderCartService

Signed-off-by: Jérémy De-Cesare <jeremy.de-cesare@corp.ovh.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#7910.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#7912.0 ✘  
* 
  * Build ✘

File: packages/manager/modules/web-universe-components/src/order-cart/order-cart.service.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+export default class {
```
Comment (cbourgois) [inline packages/manager/modules/web-universe-components/src/order-cart/order-cart.service.js]: Could you define a class name ?

Comment (JeremyDec) [reply packages/manager/modules/web-universe-components/src/order-cart/order-cart.service.js]: @cbourgois done :)

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#7954.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#7957.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#7967.0 ✘  
* 
  * Build ✘

**Comment by marie-j** [comment]
It would be nice to remove duplicated service in `web` universe

**Comment by JeremyDec** [comment]
@marie-j agree but this should be done in a separate PR, in order to not break features already merged. What do you think ?

**Comment by Jisay** [comment]
For discussion: wouldn't be nice to have a separate component for order cart management? Something like `ng-ovh-order-cart` in `packages/components` folder?

File: packages/manager/modules/web-universe-components/src/order-cart/order-cart.service.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
+export default class OrderCartService {
+  /* @ngInject */
+  constructor($q, OvhApiOrder, User) {
```
Comment (Jisay) [inline packages/manager/modules/web-universe-components/src/order-cart/order-cart.service.js]: Where does come `User`?

**Comment by frenautvh** [comment]
> For discussion: wouldn't be nice to have a separate component for order cart management? Something like `ng-ovh-order-cart` in `packages/components` folder?

i agree, cart management is not linked to web and should have a separate module just like ovh-payment-method

**Comment by JeremyDec** [comment]
@Jisay / @frenautvh In fact yes, this component should not be just for `web`, or another, as for now it is located in wuc, to allow some others to use it outside of `web`, as it is, since wuc can be used elsewhere than web. 

But, all the cart management, should be available even outside the manager. And not only with just this list of calls.

**Comment by FredericEspiau** [comment]
@JeremyDec 

If you actually do this, it would be great to have in the module a vanillaJS version and then implement it in AngularJS :)

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#8041.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#8061.0 ■   
* 
  * Build ■

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#8078.0 ✘  
* 
  * Build ✘

**Pull request merged**